tests/libwait/wait.go

```diff
@@ -39,25 +39,89 @@ import (
 	"kubevirt.io/kubevirt/tests/watcher"
 )
 
-// WaitForVMIStartOrFailed blocks until the specified VirtualMachineInstance reached either Failed or Running states
-func WaitForVMIStartOrFailed(obj runtime.Object, seconds int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	return waitForVMIPhase(ctx, []v1.VirtualMachineInstancePhase{v1.Running, v1.Failed}, obj, seconds, wp, true)
+const defaultTimeout = 360
+
+// Option represents an action that enables an option.
+type Option func(waiting Waiting)
+
+// Waiting represents the waiting struct.
+type Waiting struct {
```

Comment:
nit: I would have preferred to have here a non-verb name (e.g. waiter).

Reply:
nit: I wouldn't use waiter, might be misleading, maybe `Wait/WaitCondition`? or just `Waiting` is fine for me.

Reply:
I don't fully like the name I give to it, but as Antonio says waiter might be very misleading IMO. :)